tools/sof-dump-status.py

```diff
@@ -448,7 +448,15 @@ def dump_dapm(dapm, filter = "all"):
                 pcm['channel'] = '2'
                 pcm['dev'] = 'hw:{},{}'.format(card_id, pcm['id'])
             pipeline_lst.extend(card_info['pcm'])
-        export_pipeline(pipeline_lst)
+        try:
+            pipeline_type = list(filter(lambda x: x.strip().startswith('type'), ret_args['export'].split(' ')))[0].split(':')[1]
+        except:
+            print('Invalid filter string')
+            exit(1)
+        if pipeline_type in ['playback', 'capture']:
+            export_pipeline(list(filter(lambda x: x['type'] == pipeline_type, pipeline_lst)))
+        else:
+            export_pipeline(pipeline_lst)
```

Comment:
a comment to explain what you are doing where wouldn't hurt.

Put yourself in the shoes of a reviewer who's not a Python expert and has no time to reverse engineer your work, thanks!